development/editor/creating_icons.rst

```diff
@@ -0,0 +1,83 @@
+.. _doc_editor_icons:
+
+Editor icons
+============
+
+When a new class is created and exposed to scripting, the editor's interface
+will display it with a default icon representing the base class it's inherited
+from. Yet in most cases it is recommended to create icons for new classes
+to improve the user experience considerably.
+
+Creating icons
+~~~~~~~~~~~~~~
+
+In order to create new icons, you first need a vector graphics editor installed.
+For instance, you can use open-source `Inkscape <https://inkscape.org/>`_ editor.
+
+Clone the ``godot-design`` repository containing all the original editor icons:
+
+   .. code:: bash
+
+       git clone https://github.com/godotengine/godot-design
+
+The icons must be created in a vector graphics editor in ``svg`` format. You
+can use ``/engine/icons/inkscape_template.svg`` with default icon properties
+already set up.
+
+Once you're satisfied with the icon's design, save the icon in
+``engine/icons/svg/`` folder. But in order for the engine to automatically
+pick up the icons, each icon's filename:
+
+1. Must be prefixed with ``icon_``.
+
+2. Each consecutive capitalized letter in the class name must be converted to
+   lowercase with ``_`` suffix.
+
+So in the case of ``CSGPolygon`` class, the icon's filename should be
+``icon_c_s_g_polygon.svg``.
+
+Icon optimization
+~~~~~~~~~~~~~~~~~
+Because the editor renders the ``svg``'s at runtime, they need to be small
+in size, so they can be efficiently parsed. Editor icons must be first
+optimized before being added to the engine, to do so:
+
+1. Add them to the ``/engine/icons/svg`` folder.
+
+2. Run the ``optimize.py`` script. You must have ``scour`` package installed:
+
+   .. code:: bash
+
+       pip install scour
+       cd godot-design/engine/icons && ./optimize.py
+
+The optimized icons will be generated in ``/engine/icons/optimized`` folder.
+
+Integrating and sharing the icons
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If you're contributing to the engine itself, you should make a pull request
+to ``godot-design`` repository first. Once it's merged, make another pull
```

Comment:
*the* `godot-design` repository.

Reply:
In practice the workflow is more the other way around though, we merge first in the main godot repo if the change is wanted, and then the original version can be added to godot-design.

Reply:
I don't see a reason why "in theory" workflow should be described at all then, but perhaps "ideally" is the right word for this.